### PR TITLE
SONARCS-576 The C# sensor should also import SARIF issues from 3rd party plugins

### PR DESCRIPTION
**Pull request by dbolkensteyn**

### CODE REVIEW

File: src/main/java/org/sonar/plugins/csharp/CSharpSensor.java [hunk 1]
```diff
@@ -235,9 +232,13 @@ private void importRoslynReport(String reportPath) {
...
 
-    Set<String> activeRuleIds = Sets.newHashSet();
-    for (ActiveRule activeRule : ruleProfile.getActiveRulesByRepository(CSharpPlugin.REPOSITORY_KEY)) {
-      activeRuleIds.add(activeRule.getRuleKey());
+    ImmutableMultimap<String, ActiveRule> activeRoslynRulesByPluginKey = RoslynProfileExporter.activeRoslynRulesByPluginKey(ruleProfile.getActiveRules());
+    Map<String, String> repositoryKeyByRoslynRuleKey = Maps.newHashMap();
+    for (ActiveRule activeRoslynRule: activeRoslynRulesByPluginKey.values()) {
+      String previousRepositoryKey = repositoryKeyByRoslynRuleKey.put(activeRoslynRule.getRuleKey(), activeRoslynRule.getRepositoryKey());
+      if (previousRepositoryKey != null) {
+        throw new IllegalArgumentException("Rule keys must be unique, but \"" + activeRoslynRule.getRuleKey() + "\" is defined in both the \"" + previousRepositoryKey + "\" and \"" + activeRoslynRule.getRepositoryKey() + "\" rule repositories.");
```
Comment (SonarTech) [inline src/main/java/org/sonar/plugins/csharp/CSharpSensor.java]: MINOR Split this 246 characters long line (which is greater than 180 authorized). [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS00103)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MINOR 1 minor

Watch the comments in this conversation to review them.

File: src/main/java/org/sonar/plugins/csharp/RoslynProfileExporter.java [hunk 2]
```diff
@@ -138,10 +138,14 @@ public void exportProfile(RulesProfile rulesProfile, Writer writer) {
     appendLine(writer, "</RoslynExportProfile>");
   }
 
-  private static ImmutableMultimap<String, ActiveRule> activeRoslynRulesByPluginKey(List<ActiveRule> activeRules) {
+  public static ImmutableMultimap<String, ActiveRule> activeRoslynRulesByPluginKey(List<ActiveRule> activeRules) {
     ImmutableMultimap.Builder<String, ActiveRule> builder = ImmutableMultimap.builder();
 
     for (ActiveRule activeRule : activeRules) {
+      if (activeRule.getRule().getLanguage() != CSharpPlugin.LANGUAGE_KEY) {
```
Comment (SonarTech) [inline src/main/java/org/sonar/plugins/csharp/RoslynProfileExporter.java]: MAJOR Change this comparison to use the equals method. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1698)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Comment by dbolkensteyn** [comment]
The filtering to only import C# issues isn't working in SonarQube, I'm closing this PR in favor of https://github.com/SonarSource/sonar-csharp/pull/55

**Pull request closed without merging**